### DISCUSSION TITLE
Time to try S7?

### DISCUSSION THREAD

**Discussion by JamesHWade** [MichelNivard/gptstudio#137] [Ideas]
@calderonsamuel, I'll need your advice on this one.

What do you think about using an S7 class as a re-implementation of the R6 class you used to get streaming to work? You were right with your original design, I think. I want to move away from the need to save to disk since that's creating other issues and limitations.

Does S7 makes sense?

**Comment by calderonsamuel** [comment]
I believe S7 makes more sense for the (build -> translate -> perform) process because they are both functional OOP systems that aim to be used in the global environment. On this front, we could decide whether or not we would like to be part of the, still maturing, S7 experiment. Could be fun!

On the R6 matter, I agree that streaming *in memory* is better than doing it *in disk*. We could think of something that combines the encapsulating power of R6 + some kind of parallel processing for the realtime communication between the R's streaming process and the JS's HTML rendering process. In the previous approach, I was using the shiny's `session$sendCustomMessage()` method inside another R6 method, which depended on a CSS class generated by the `streamingMessage()` widget. All this made very hard to understand the whole picture (and it wasn't documented anywhere). 

Having said that, we could revert to that approach, with better documentation, while we think in how to improve it. I would like to know the issues or limitations you have encountered in the *in disk* implementation, so we avoid falling into them again.
